src/app/modules/main/wallet_section/send/view.nim

```diff
@@ -340,6 +340,14 @@ QtObject:
       SendType(sendType), lockedInAmounts)
   
   proc authenticateAndTransferWithParameters*(self: View, uuid: string, accountFrom: string, accountTo: string, token: string, toToken: string,
-    sendType: int, tokenName: string, tokenIsOwnerToken: bool, rawPaths: string) {.slot.} =
+    sendTypeInt: int, tokenName: string, tokenIsOwnerToken: bool, rawPaths: string, slippagePercentageString: string) {.slot.} =
+
+    let sendType = SendType(sendTypeInt)
+
+    var slippagePercentage: Option[float]
```

Comment:
why is this Option[float] required here? we get a string of type float from backend and we just convert it? or maybe I don't understand why we need an optional variable at this stage

Reply:
I made it an `Option` since it only makes sense for some Tx types (Swap only for now). For all other the caller should enter an empty string.
Another, perhaps more reasonable, possibility would be to have different functions for each Tx type, each with the appropriate parameters, so the caller doesn't have to fill so many unneeded parameters. Anyway, that could be part of the rework mentioned before.